Makefile

```diff
@@ -176,6 +186,12 @@ out/minikube_$(DEB_VERSION).deb: out/minikube-linux-amd64
 	dpkg-deb --build out/minikube_$(DEB_VERSION)
 	rm -rf out/minikube_$(DEB_VERSION)
 
+out/minikube-all-amd64.tar.gz: pkg/minikube/assets/assets.go out/localkube out/minikube-linux-amd64 out/minikube-darwin-amd64 out/minikube-windows-amd64.exe
+	tar czvf out/minikube-all-amd64.tar.gz $($(shell echo TAR_TARGETS_ALL | tr a-z A-Z))
+
+out/minikube-%-amd64.tar.gz: pkg/minikube/assets/assets.go out/minikube-$(GOOS)-$(GOARCH)$(IS_EXE)
```

Comment:
this rule will actually cover both since % will match the "all".

I envisioned the `TAR_TARGET_ALL` as the shared files between the two, like addons, etc.  How would we use the "all" bundle with everything in it?

I'm not sure if the wildcard matching will work with this one, since you'll have conditional targets
Maybe something like this would work

 I dont think you can use automatic variables like $* in the target section.  This is maybe a hack to get the DAG right

```
out/minikube-%-amd64.tar.gz:
	$(MAKE) $($(shell echo TAR_TARGETS_$* | tr a-z A-Z))
	tar -cvf $@ $($(shell echo TAR_TARGETS_$* | tr a-z A-Z))
```

The whole `shell echo...` bit is somewhat of a hack to evaluate OS-specific variables, by taking the pattern matched "darwin", "linux", "windows", casting it to uppercase and appending it to the end of `TAR_TARGETS_`

You can't really rely on $(GOOS) and $(IS_EXE) since we might be cross compiling